src/sentry/api/authentication.py

```diff
@@ -191,42 +246,30 @@ def authenticate_credentials(self, relay_id, relay_sig, request):
 @AuthenticationSiloLimit(SiloMode.CONTROL, SiloMode.REGION)
 class ApiKeyAuthentication(QuietBasicAuthentication):
     token_name = b"basic"
+    _hybrid_cloud_rollout_level = 1
 
     def accepts_auth(self, auth: list[bytes]) -> bool:
         return bool(auth) and auth[0].lower() == self.token_name
 
-    def _authenticate_credentials(self, userid):
-        key = ApiKeyReplica.objects.filter(key=userid).last()
-        if key is None:
-            raise AuthenticationFailed("API key is not valid")
-        if not key.is_active:
-            raise AuthenticationFailed("Key is disabled")
-        with configure_scope() as scope:
-            scope.set_tag("api_key", key.apikey_id)
-        return AuthenticatedToken.from_token(key)
-
     def authenticate_credentials(self, userid, password, request=None):
         # We don't use request, but it needs to be passed through to DRF 3.7+.
         if password:
             return None
 
-        if SiloMode.get_current_mode() == SiloMode.REGION or options.get(
-            "hybrid_cloud.authentication.use_api_key_replica"
-        ):
-            return AnonymousUser(), self._authenticate_credentials(userid)
-
-        try:
-            key = ApiKey.objects.get_from_cache(key=userid)
-        except ApiKey.DoesNotExist:
-            raise AuthenticationFailed("API key is not valid")
+        if SiloMode.get_current_mode() == SiloMode.REGION:
+            key = ApiKeyReplica.objects.filter(key=userid).last()
+            if key is None:
+                raise AuthenticationFailed("API key is not valid")
```

Comment:
Would there be any value in moving this to conditional model lookup to the auth service impl instead?